assets/js/File_Zipper_js/huffman.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Created by aarnavjindal on 25/04/20.
+ */
+
+import { BinaryHeap } from './heap.js';
+
+export { HuffmanCoder }
+
+class HuffmanCoder{
+
+    stringify(node){
+        if(typeof(node[1])==="string"){
+            return '\''+node[1];
+        }
+
+        return '0' + this.stringify(node[1][0]) + '1' + this.stringify(node[1][1]);
+    }
+
+    display(node, modify, index=1){
+        if(modify){
+            node = ['',node];
+            if(node[1].length===1)
+                node[1] = node[1][0];
+        }
+
+        if(typeof(node[1])==="string"){
+            return String(index) + " = " + node[1];
+        }
+
+        let left = this.display(node[1][0], modify, index*2);
+        let right = this.display(node[1][1], modify, index*2+1);
+        let res = String(index*2)+" <= "+index+" => "+String(index*2+1);
+        return res + '\n' + left + '\n' + right;
+    }
+
+    destringify(data){
+        let node = [];
+        if(data[this.ind]==='\''){
+            this.ind++;
+            node.push(data[this.ind]);
+            this.ind++;
+            return node;
+        }
+
+        this.ind++;
+        let left = this.destringify(data);
+        node.push(left);
+        this.ind++;
+        let right = this.destringify(data);
+        node.push(right);
+
+        return node;
+    }
```

Comment:
Can you please check this function again? I am get the following error:
```
Uncaught RangeError: Maximum call stack size exceeded
```
I believe this is happening because of excessive recursive calls. Please look into it.